### ISSUE TITLE
grid documentation: add these useful pieces of info in the right place please

### ISSUE THREAD

**Issue by DieterRogiest** [SeleniumHQ/selenium#6747]
This is about:
For nodes (and hubs?) to use the right network interfaces (on Linux), I had to remove docker and virbr0 interfaces.
Sometimes UPDATE: no, NOT sometimes (I just figured out that I have a chromedriver and a geckodriver in my PATH since 3 months), nodes need to be started with specific webdrivers (java -Dwebdriver.chrome.driver=./chromedriver -Dwebdriver.firefox.driver=./geckodriver -jar ...)
FirefoxOptions() sets marionette default to false (in my experience).

Learning selenium is a bit frustrating because of bad and not up-to-date documentation lying in all corners of the internet.

For selenium grid there is
https://www.seleniumhq.org/docs/07_selenium_grid.jsp
and then later on you stumble somewhere on https://github.com/SeleniumHQ/selenium/wiki/Grid2 (there is also https://github.com/SeleniumHQ/selenium/wiki/Grid-Platforms)
and via https://github.com/SeleniumHQ/selenium I now also discover New Handbook (work in progress): https://seleniumhq.github.io/docs/ (still have to read that)

I learned the java way of selenium.
Grid (remotewebdriver) I got working ok locally on my macbook laptop: both normal and docker-selenium.
The normal local grid using only selenium-server-standalone-3.141.59.jar.
I figured that that standalone jar contains a chromedriver and a geckodriver (no idea what other drivers it contains), in the output of my maven command I can see "Starting ChromeDriver 2.42.591059", which was not the latest version of chromedriver.
For gecko firefox the output does not show what version it was.
**UPDATE: a few hours later I see that I have a chromedriver and a geckodriver in my PATH on my macbook since 3 months, I completely forgot about that. So no, the standalone jar does NOT contain drivers.** My apologies

But also I found that the local node can be started with 
UPDATE: no, it's mandatory
java -Dwebdriver.chrome.driver=./chromedriver -Dwebdriver.firefox.driver=./geckodriver\ 2 -jar ...
and indeed using browsertype/OS specific webdrivers can also be done on the node: the output then showed "Starting ChromeDriver 2.44.609545".

This info is not on 07_selenium_grid.jsp or on wiki/Grid2 (it is mentioned somehow in the New Handbook) and it's also not in the help output of "java -jar selenium-server-standalone-3.141.59.jar -role node -h".  
 
For docker-selenium hub and node containers I don't know if the nodes can be started with specific webdrivers (I guess you would have to build compose a docker image yourself to have the specicic webdriver file in its file system and let it start with the -Dwebdriver.<browsertype>.driver= option but that's too much hassle)

For firefox I got the error:
Error forwarding the new session cannot find : Capabilities {acceptInsecureCerts: true, browserName: firefox, marionette: false, moz:firefoxOptions: {args: [], prefs: {}}}

I got it working by setting in the test code the capability of marionette to true (so FirefoxOptions() sets it default to false?):

```
        FirefoxOptions options = new FirefoxOptions();
        options.setCapability("marionette", true);
        driver = new RemoteWebDriver(
                new URL(Hub),
                options);
```

Then it took a lot of work getting a non-local grid working at home:
The issues were the network configuration on my 2 node linux machines (real ones, not VMs):
In the output of the hub I saw that my node Kubuntu 18.04 used a docker network interface. I stopped and disabled docker and rebooted so that interface would not show up anymore in ifconfig.
Then it used the virbr0 network interface (the nat interface for kvm): another 3 commands to get rid of that in ifconfig. (I don't know yet how to get it back if I want to use kvm again)
Then that node finally used the correct ip address 192.168.0.xx

On my node Mageia 6 I hit a bug of Mageia when I disabled the firewall but which actually  firewalled everything coming in. After enabling the firewall again and puching a hole in port 5555 that node started to function.

Next issue was:
on the Mageia node the included chromedriver of the selenium-server-standalone-3.141.59.jar could be used with chrome beta 71.0.3578.80 and the chrome test succeeded.
UPDATE: no, there is no included chromedriver, I probably just have also on Mageia a chromedriver in its PATH.
My firefox test didn't start the firefox browser, I guess because on Mageia the firefox version is 60.something.ESR. I didn't try the latest firefox on it.

But with the Kubuntu node I got in my maven output the errors:

Unable to create new service: ChromeDriverService
Driver info: driver.version: unknown
Unable to create new service: GeckoDriverService
Driver info: driver.version: unknown

I got it working by starting the node with ... -Dwebdriver.chrome.driver=./chromedriver -Dwebdriver.firefox.driver=./geckodriver ... (these linux webdrivers in the same location as the selenium-server-standalone-3.141.59.jar of the node).
Then on Kubuntu the chrome and firefox tests worked.

No idea why only on Kubuntu I needed to have specific webdrivers.
UPDATE: no, the Mageia probably has a chromedriver in its PATH that I forgot about.
My Mageia had chrome beta 71.0.3578.80
My Kubuntu had chrome beta 71.0.3578.53 (and firefox 63.0.3)

**Comment by DieterRogiest** [comment]
My apologies, a few hours later I find out that on my macbook I have chromedriver and geckodriver in my PATH (in /usr/local/bin/ ) since 3 months that I completely forgot about. I added UPDATE lines to my text above.
Now this bug is only about that I had to remove docker and kvm network interfaces, that marionette options is default false, and that it's hard to find on the internet that nodes need to be started with specific webdrivers.
